graphviz/files.py

```diff
@@ -143,7 +144,14 @@ def save(self, filename=None, directory=None):
         if directory is not None:
             self.directory = directory
 
-        filepath = self.filepath
+        if self.filename is None and self.directory is None and cleanup:
```

Comment:
`self.filename` is never `None` (see `__init__()`) so this branch never gets executed.
Still, I think using a `tempfile` should either be triggered by a dedicated flag or a special value for `filename` (i.e. it should be orthogonal to `cleanup`).